app/javascript/js/controllers/tabs_controller.js

```diff
@@ -70,4 +79,141 @@ export default class extends Controller {
+
+      if (isActive || isFocused) {
+        button.setAttribute('tabindex', '0')
+      } else {
+        button.setAttribute('tabindex', '-1')
+      }
+
+      this.toggleTabClasses(button, isActive)
+
+      const wrapper = button.closest('.tabs__item-wrapper--scope')
+      if (wrapper) {
+        this.toggleTabClasses(wrapper, isActive, button)
+      }
+    })
+  }
+
+  toggleTabClasses(element, isActive, sourceElement = element) {
+    const activeClass = sourceElement.dataset.tabActiveClass
+    const inactiveClass = sourceElement.dataset.tabInactiveClass
+
+    if (activeClass) element.classList.toggle(activeClass, isActive)
+    if (inactiveClass) element.classList.toggle(inactiveClass, !isActive)
+  }
+
+  handleKeyDown(e) {
+    const currentTab = e.currentTarget
+    const tabs = this.getEnabledTabs()
+    const currentIndex = tabs.indexOf(currentTab)
+
+    if (currentIndex === -1) return
+
+    let targetTab = null
+    let shouldPreventDefault = false
+
+    switch (e.key) {
+      case 'ArrowRight':
+      case 'ArrowDown':
+        targetTab = tabs[(currentIndex + 1) % tabs.length]
+        shouldPreventDefault = true
+        break
+
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        targetTab = tabs[(currentIndex - 1 + tabs.length) % tabs.length]
+        shouldPreventDefault = true
+        break
+
+      case 'Home':
+        targetTab = tabs[0]
+        shouldPreventDefault = true
+        break
+
+      case 'End':
+        targetTab = tabs[tabs.length - 1]
+        shouldPreventDefault = true
+        break
+
+      case 'Enter':
+      case ' ':
+        if (currentTab.dataset.tabId) {
+          this.activateTab(currentTab)
+          shouldPreventDefault = true
+        }
+        break
+    }
+
+    if (shouldPreventDefault) {
+      e.preventDefault()
+    }
+
+    if (targetTab) {
+      this.focusTab(targetTab)
+    }
+  }
+
+  getEnabledTabs() {
+    return this.tabButtonTargets.filter((button) => {
+      const disabled =        button.getAttribute('aria-disabled') === 'true'
+        || button.hasAttribute('disabled')
+
+      return !disabled
+    })
+  }
+
+  focusTab(tabButton) {
+    // Remove focus class from all tabs first
+    this.removeKeyboardFocusClass()
+
+    // Add keyboard focus class and focus the tab
+    tabButton.classList.add('tabs__item--focused')
+    tabButton.setAttribute('tabindex', '0')
+    tabButton.focus()
+  }
+
+  removeKeyboardFocusClass() {
+    // Remove keyboard focus class from all tabs
+    this.tabButtonTargets.forEach((button) => {
+      button.classList.remove('tabs__item--focused')
+    })
+  }
+
+  // Handle blur event - remove focus class when tab loses focus
+  handleBlur(e) {
+    e.currentTarget.classList.remove('tabs__item--focused')
+  }
+
+  // Handle mousedown - remove focus class immediately on mouse interaction
+  handleMouseDown(e) {
+    // Remove focus class when mouse is used (prevents focus ring on click)
+    this.removeKeyboardFocusClass()
+  }
```

Comment:
Wondering the same about this one... Do we actually need this? Genuinelly asking.